src/mongo_odm/deserializing_cursor.hpp

```diff
@@ -85,11 +89,38 @@ class deserializing_cursor<T>::iterator : public std::iterator<std::input_iterat
      * underlying collection cursor iterator.
      */
     T operator*() {
-        return bson_mapper::to_obj<T>(*_ci);
+        return _opt ? _opt.value() : bson_mapper::to_obj<T>(*_ci);
```

Comment:
How can `_opt` be unset here? It seems like either we will have a cached value or the cursor will be empty.